src/aws-manager.js

```diff
@@ -824,9 +825,18 @@ class AwsManager {
     }
 
     await Promise.all(worker.regions.map(async r => {
+      let allSGForRegion = [];
       await Promise.all(worker.instanceTypes.map(async t => {
         let launchSpec = launchSpecs[r.region][t.instanceType].launchSpec;
 
+        if (launchSpec.SecurityGroups) {
+          for (let sg of launchSpec.SecurityGroups) {
+            if (!_.includes(allSGForRegion, sg)) {
+              allSGForRegion.push(sg);
+            }
+          }
+        }
```

Comment:
i know, i usually do that.  On second thought, mutating things outside of a map in the map's callback is a bad idea just in general :)

let me agree more aggressively!